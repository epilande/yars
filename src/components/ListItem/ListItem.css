.item {
  padding: 1rem;
  border-bottom: 1px solid color(#433a5a alpha(25%));
  background: #eef3f5;
  color: var(--dark);

  &:hover .remove {
    opacity: 1;
  }
}

.itemText {
  width: calc(100% - 4rem);
  display: inline-block;
  outline: none;
  letter-spacing: normal;
}

.checkbox {
  position: relative;
  display: inline-block;

  input {
    margin-right: 1rem;
    outline: 0;
    border: 1px solid color(#433a5a alpha(25%));
    border-radius: 1px;
    width: 1rem;
    height: 1rem;
    vertical-align: middle;
    appearance: none;
  }

  &::after {
    content: '\2713';
    position: absolute;
    top: -0.25rem;
    left: 0.0625rem;
    opacity: 0;
    color: color(#433a5a alpha(75%));
    font-size: 1.3rem;
    cursor: pointer;
  }
}

.remove {
  opacity: 0;
  float: right;
  transition: opacity 0.2s;
  cursor: pointer;

  &:hover {
    color: var(--secondary);
    font-weight: 600;
  }
}

.completed {
  .itemText {
    text-decoration: line-through;
    opacity: 0.5;
  }

  .checkbox {
    input {
      border-color: color(#433a5a alpha(40%));
    }

    &::after {
      opacity: 1;
    }
  }
}
