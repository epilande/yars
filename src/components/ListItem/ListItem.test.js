import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';

describe('ListItem', () => {
  it('should render a ListItem', () => {
    const wrapper = shallow(
      <ListItem
        item={{ id: 1, text: 'Hello' }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
