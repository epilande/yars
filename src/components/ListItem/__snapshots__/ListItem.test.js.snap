exports[`ListItem should render a ListItem 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <li
    className="item">
    <div
        className="itemText"
        onClick={[Function]}>
        Hello
    </div>
</li>,
  "nodes": Array [
    <li
      className="item">
      <div
            className="itemText"
            onClick={[Function]}>
            Hello
      </div>
</li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ListItem
        item={
                Object {
                        "id": 1,
                        "text": "Hello",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ListItem {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onBlur": [Function],
        "onChange": [Function],
        "onCheck": [Function],
        "onEnter": [Function],
        "onFocus": [Function],
        "onRemove": [Function],
        "onTextClick": [Function],
        "props": Object {
          "item": Object {
            "id": 1,
            "text": "Hello",
          },
        },
        "refs": Object {},
        "state": Object {
          "editing": false,
          "text": "Hello",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          className="item">
          <div
                    className="itemText"
                    onClick={[Function]}>
                    Hello
          </div>
</li>,
        "_debugID": 2,
        "_renderedOutput": <li
          className="item">
          <div
                    className="itemText"
                    onClick={[Function]}>
                    Hello
          </div>
</li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ListItem
    item={
        Object {
            "id": 1,
            "text": "Hello",
          }
    } />,
}
`;
